### PR TITLE
add new check for pipe per line 

### PR DESCRIPTION
**Pull request by rands0n**
Some style guide proposes:

```elixir
# not preferred

1 |> Integer.to_string() |> String.to_integer()

# preferred

1
|> Integer.to_string()
|> String.to_integer()
```

This PR provides a check for that.

--

Credits by @bamorim

### CODE REVIEW

**Review by rrrene — CHANGES_REQUESTED** [review]
This seems off to a great start. See comments below :+1:

File: lib/credo/check/readability/pipe_per_line.ex [hunk 1]
```diff
@@ -0,0 +1,47 @@
+defmodule Credo.Check.Readability.PipePerLine do
+  use Credo.Check,
+    base_priority: :high,
```
[the remark this answers is not in the document]
Comment (rands0n) [reply lib/credo/check/readability/pipe_per_line.ex:3]: Which priority do you recommend?

File: lib/credo/check/readability/pipe_per_line.ex [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        trigger: "|>"
+      )
+    end)
+  end
+
+  defp filter_pipes({:arrow_op, _, :|>}), do: true
+  defp filter_pipes(_), do: false
+
+  defp filter_tokens({_, [_]}), do: false
+  defp filter_tokens({_, [_ | _]}), do: true
```
Comment (bamorim) [inline lib/credo/check/readability/pipe_per_line.ex:59]:
```suggestion
  defp filter_tokens({_, [_ | _]}), do: true
  defp filter_tokens(_), do: false
```

File: lib/credo/check/readability/pipe_per_line.ex (hunk 1 above)
[the remark this answers is not in the document]
Comment (rrrene) [reply lib/credo/check/readability/pipe_per_line.ex:3]: I was just asking for your reasons. If you have none, just remove `:base_priority` and the check gets the "standard" priority between high and low 👍

[the remark this answers is not in the document]
Comment (rands0n) [reply lib/credo/check/readability/pipe_per_line.ex]: Removed 🥇 

Thanks

**Comment by rands0n** [comment]
Any news?

**Comment by rrrene** [comment]
This looks good. 

The last thing I am wondering is if we covered all edge cases in the tests though. 

Are there things we could be missing? Like "exotic" combinations of single line and multi line pipes?

```elixir
String.to_integer(1 |> Integer.to_string()) 
|> Integer.to_string()
```

**Comment by bamorim** [comment]
@rrrene that is a good point. The current approach just count pipes per line, so something like

```elixir
value
|> do_something(:a |> to_string())
|> do_something_else()
```

Would fail. If we want this to be accepted, we would need to change. If we don't mind that much, we could suggest that this is changed either to an extracted variable or to something like

```elixir
value
|> do_something(
  :a
  |> to_string()
)
|> do_seomthing_else()
```

**Comment by bamorim** [comment]
The implementation to support that case would require us parsing the AST and counting `|>` per line per tree depth, which is harder to implement and I'm not sure worth it.

**Comment by rands0n** [comment]
@rrrene ping

**Comment by rrrene** [comment]
This is part of Credo [v1.7.0-rc.1](https://hex.pm/packages/credo/1.7.0-rc.1) :+1:
